### PR TITLE
allow input connectors to multiple components connector

### PR DESCRIPTION
**Pull request by arun3688**
### Related Issues

https://github.com/OpenModelica/OMSimulator/issues/919

### Purpose

This PR allows` input connector` to be connected with `multiple component input connector` which has no previous connections defined.  According to `SSP-1.0 ` specification multiple connections to already connected  connectors of type input/output/parameter should not exist.

### CODE REVIEW

**Review by lochel — COMMENTED** [review]
Haven't read the code yet, but I can confirm that it seems to work:
image

Just the error message in case of connecting output connectors several times needs to be improved. I propse something like:
> [addConnection] Connector y is already connected to addSat.y

File: testsuite/api/connections.lua [hunk 1]
```diff
@@ -90,8 +90,7 @@ printStatus(status, 0)
 -- status:  [correct] error
 -- error:   [addConnection] Type mismatch in connection: sc1.u2 -> sc2.y2
 -- status:  [correct] error
--- error:   [addConnection] Connector is already connected: sc1.u1
--- status:  [correct] error
+-- status:  [wrong] ok
```
Comment (lochel) [inline testsuite/api/connections.lua:93]: @arun3688 This looks wrong.

File: testsuite/api/connections.py [hunk 2]
```diff
@@ -90,8 +90,7 @@ def printStatus(status, expected):
 ## status:  [correct] error
 ## error:   [addConnection] Type mismatch in connection: sc1.u2 -> sc2.y2
 ## status:  [correct] error
-## error:   [addConnection] Connector is already connected: sc1.u1
-## status:  [correct] error
+## status:  [wrong] ok
```
Comment (lochel) [inline testsuite/api/connections.py:93]: @arun3688 and this as well - but should be the same issue as the one above.

Comment (arun3688) [reply testsuite/api/connections.py:93]: @lochel this should be fixed with this https://github.com/OpenModelica/OMSimulator/pull/925/commits/d646dbc5ac9c9c687298960bcdd2617225d7d9e6

File: testsuite/api/connections.lua (hunk 1 above)
Comment (arun3688) [reply testsuite/api/connections.lua]: @lochel this should be fixed with this d646dbc

**Review by lochel — APPROVED** [review]
Nice! 🚀

**Comment by lochel** [comment]
The error messages are relevant and helpful:

image

**Pull request merged**